### PR TITLE
Greeting Message Hook

### PR DESCRIPTION
**Pull request by NikolasGraham**

### CODE REVIEW

File: src/hooks/useGreetingMessage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    const greetingStyles = [
+        // 0 Standard Greeting
+        `Hello ${name}, welcome back!`,
+
+        // 1 - English Greeting
+        `Ello govenor ${name}`,
+
+        // 2 - Irish Greeting
+        `Top of the morning to you ${name}, how are you today lad?`
```
Comment (toha87) [inline src/hooks/useGreetingMessage.js:22]: THis is funny :)

File: src/hooks/useGreetingMessage.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        // 3 - Scottish Greeting
+
+        // n - .......
+    ];
+
+    // Randomized right now
+    // Has potential to select specific greetings
+    // User can add own?
+    useEffect(() => {
+        setStyle(Math.floor(Math.random() * Math.floor(greetingStyles.length)));
```
Comment (toha87) [inline src/hooks/useGreetingMessage.js]: 👍

File: src/containers/Home/Header.js [hunk 3]
```diff
@@ -3,17 +3,27 @@
...
 
 const Header = () => {
 
     const context = useContext(AppContext);
 
     const profileHook = useProfile();
+    const greetingHook = useGreetingMessage();
+
+    useEffect(() => {
+        if(context.mongoHook.firstName !== ""){
```
Comment (toha87) [inline src/containers/Home/Header.js]: Maybe another check for the last name?

File: src/hooks/useGreetingMessage.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    const voiceContext = useContext(VoiceCommandsContext);
+    const loggingContext = useContext(LoggingContext).logger;
+
+    const [style, setStyle] = useState(0);
+    const [name, setName] = useState("");
+
+    const greetingStyles = [
+        // 0 Standard Greeting
+        `Hello ${name}, welcome back!`,
```
Comment (toha87) [inline src/hooks/useGreetingMessage.js:16]: maybe a null check for the name. As I understand you're setting name outside of the hook

File: src/hooks/useMongo.js [hunk 5]
```diff
@@ -148,6 +162,8 @@ export const useMongo = (input, logger) => {
     const logout = () => {
         setAuthenticatedUser({});
         setIsLoggedIn(false);
+        setFirstName("");
+        setLastName("");
```
Comment (toha87) [inline src/hooks/useMongo.js:169]: This is good!

File: src/hooks/useGreetingMessage.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+        // 2 - Irish Greeting
+        `Top of the morning to you ${name}, how are you today lad?`
+
+        // 3 - Scottish Greeting
+
+        // n - .......
+    ];
+
+    // Randomized right now
```
Comment (blazinaj) [inline src/hooks/useGreetingMessage.js:36]: I love this, so much!!!

File: src/hooks/useGreetingMessage.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }, [name]);
+
+    return{
+        style,
+        setStyle,
+        greetingStyles,
+        setName
+    }
+
```
Comment (blazinaj) [inline src/hooks/useGreetingMessage.js]: Really well done creating a custom hook. You implemented state, useEffect, context, and hook chaining beautifully. These are advanced techniques that have taken me over a year to figure out. You should be proud of this work.

**Review by blazinaj — APPROVED** [review]

**Pull request merged**